### PR TITLE
use reformatter.el (#1)

### LINKED ISSUE #1: maintain approximate position in buffer
Thanks for this elisp support! I thought I would want the before-save functionality, but what I found disappointing was that it jumps to the beginning of the buffer each time, causing me to lose context.

I realize (after I tried it) that the fix isn't as simple as adding a save-excursion, because it replaces the whole buffer. And it isn't as simple as restoring by line number because that can change substantially. But I wonder if there's a way to maintain the approximate position based on content… is it something you've thought about?

### PR DESCRIPTION
**Pull request by amesgen**
Closes #1.

### CODE REVIEW

**Comment by mrkkrp** [comment]
Does it work for you? @amesgen for me `reformatter` just hangs with `ormolu`. I do not know why.

**Comment by amesgen** [comment]
Yes, this branch is working fine for me, no problems.

**Comment by vyorkin** [comment]
Thank you! I have just a few notes/suggestions:

File: ormolu.el [hunk 1]
```diff
@@ -42,74 +43,23 @@
...
-  :keymap ormolu-mode-map
-  :lighter " Or"
+  "Local keymap used for `ormolu-format-on-save-mode`.")
+
+;;;###autoload (autoload 'ormolu-format-buffer "ormolu" nil t)
+;;;###autoload (autoload 'ormolu-format-region "ormolu" nil t)
+;;;###autoload (autoload 'ormolu-format-on-save-mode "ormolu" nil t)
+(reformatter-define ormolu-format
+  :program ormolu-process-path
+  :args ormolu-extra-args
```
Comment (vyorkin) [inline ormolu.el]: For me, it doesn't work without specifying a source file to format:

```suggestion
  :args (append ormolu-extra-args '("/dev/stdin"))
```

**Review by vyorkin — CHANGES_REQUESTED** [review]

File: ormolu.el [hunk 2]
```diff
@@ -42,74 +43,23 @@
   :safe #'stringp)
 
 (defcustom ormolu-extra-args nil
```
Comment (vyorkin) [inline ormolu.el]:
```suggestion
(defcustom ormolu-extra-args '()
```

File: ormolu.el (hunk 1 above)
Comment (amesgen) [reply ormolu.el]: Hm, weird, works just fine for me. Will change though.

@mrkkrp Maybe this is the reason why it didn't work for you?

**Review by vyorkin — APPROVED** [review]

**Pull request merged**